label_studio/ml/models.py

```diff
@@ -87,45 +123,110 @@ class MLBackend(models.Model):
     )
     created_at = models.DateTimeField(_('created at'), auto_now_add=True)
     updated_at = models.DateTimeField(_('updated at'), auto_now=True)
+
+    # deprecated
     auto_update = models.BooleanField(
         _('auto_update'),
         default=True,
         help_text='If false, model version is set by the user, if true - getting latest version from backend.',
     )
 
-    def __str__(self):
-        return f'{self.title} (id={self.id}, url={self.url})'
+    @staticmethod
+    def healthcheck_(url, auth_method=None, **kwargs):
+        return MLApi(url=url, auth_method=auth_method, **kwargs).health()
 
     @staticmethod
-    def healthcheck_(url):
-        return MLApi(url=url).health()
+    def setup_(url, project, auth_method=None, **kwargs):
+        api = MLApi(url=url, auth_method=auth_method, **kwargs)
 
-    def has_permission(self, user):
-        user.project = self.project  # link for activity log
-        return self.project.has_permission(user)
+        if not isinstance(project, Project):
+            project = Project.objects.get(pk=project)
+
+        return api.setup(project, **kwargs)
 
     @staticmethod
-    def setup_(url, project, model_version=None):
-        api = MLApi(url=url)
+    def get_versions_(url, project, auth_method, **kwargs):
+        """ """
+        api = MLApi(url=url, auth_method=auth_method, **kwargs)
+
         if not isinstance(project, Project):
             project = Project.objects.get(pk=project)
-        return api.setup(project, model_version=model_version)
+
+        return api.get_versions(project)
+
+    def __str__(self):
+        return f'{self.title} (id={self.id}, url={self.url})'
+
+    def __init__(self, *args, **kwargs):
+        super(MLBackend, self).__init__(*args, **kwargs)
+        self.__original_title = self.title
+
+    def save(self, *args, **kwargs):
+        """
+        Overrides the save() method to update the associated project's model_version field.
+
+        If the title of the model instance is changed and the model_version
+        of the related project is currently the same as the original title,
+        the project's model_version is updated to the new title.
+        """
+        p = self.project
+
+        if self.title != self.__original_title and p.model_version == self.__original_title:
+            with transaction.atomic():
+                p.model_version = self.title
+                p.save()
```

Comment:
this should also use update_fields as recommended in https://github.com/HumanSignal/label-studio/pull/5570/files#r1522245983